avm/res/edge/site/main.bicep

```diff
@@ -0,0 +1,100 @@
+targetScope = 'subscription'
+
+// This module supports both subscription and resource group scope deployments
+// Use the deploymentScope parameter to specify which scope to use
+metadata name = 'Microsoft Edge Site'
+metadata description = 'This module deploys a Microsoft Edge Site at subscription or resource group scope.'
+
+// ============== //
+//   Parameters   //
+// ============== //
+
+@description('Optional. The scope at which to deploy the module. Valid values are "subscription" or "resourceGroup".')
+@allowed(['subscription', 'resourceGroup'])
+param deploymentScope string = 'resourceGroup'
+
+@description('Required. Name of the resource to create.')
+param name string
+
+@description('Required. Location for all Resources.')
+param location string
+
+@description('Conditional. Name of the resource group. Required if deploymentScope is "resourceGroup".')
+param resourceGroupName string?
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. The description of the site.')
+param siteDescription resourceInput<'Microsoft.Edge/sites@2025-03-01-preview'>.properties.description?
```

Comment:
I'd strongly advice to only use resource-derived types for arrays of objects or objects. Otherwise they don't add value, but actually have the drawback of making it impossible to understand what type is actually expected from looking at the template or the readme.
So only when referencing the module you'd be able to see what's actually expected.

So in this case, for example I assume? the `siteDescription`  is a `string` and hence the line should say
`param siteDescription string?`